### PR TITLE
RFC: Spike on associations

### PR DESCRIPTION
**Pull request by ianks**
So I made a quick spike for implementing associations in rom-http to see how it might work. A couple of things I have questions about, to see if this can be ironed out. Check the comments :arrow_down:

### CODE REVIEW

File: examples/repository_with_combine.rb [hunk 1]
```diff
@@ -1,12 +1,19 @@
...
 require 'json'
 require 'uri'
 require 'net/http'
 require 'rom-repository'
+require 'rom/http'
+
+module ROM
+  class Relation
+    class Loaded
+      def struct_namespace(*args)
```
Comment (ianks) [inline examples/repository_with_combine.rb:11]: For some reason, this was needed...

File: examples/repository_with_combine.rb [hunk 2]
```diff
@@ -15,25 +22,34 @@ def call(dataset)
...
     if %i(post put patch).include?(dataset.request_method)
-      JSON.parse(response.body, symbolize_names: true)
+      bleh(JSON.parse(response.body, symbolize_names: true))
     else
-      Array([JSON.parse(response.body, symbolize_names: true)]).flatten
+      Array([JSON.parse(response.body, symbolize_names: true)]).flatten.map(&method(:bleh))
     end
   end
+
+  def bleh(item)
```
Comment (ianks) [inline examples/repository_with_combine.rb:46]: attribute aliases do not seem to work properly with rom-http for some reason, so this is a hack

File: examples/repository_with_combine.rb [hunk 3]
```diff
@@ -54,32 +75,34 @@ def by_id(id)
 class Posts < ROM::Relation[:http]
   schema(:posts) do
     attribute :id, ROM::Types::Integer.meta(primary_key: true)
-    attribute :userId, ROM::Types::Integer, alias: :user_id
+    attribute :user_id, ROM::Types::Integer.meta(foreign_key: true)
```
Comment (ianks) [inline examples/repository_with_combine.rb:78]: how can we make aliases work?

File: examples/repository_with_combine.rb [hunk 4]
```diff
@@ -44,8 +60,13 @@ class Users < ROM::Relation[:http]
     attribute :email, ROM::Types::String
     attribute :phone, ROM::Types::String
     attribute :website, ROM::Types::String
+
+    associations do
+      has_many :posts, view: :for_user, override: true
```
Comment (ianks) [inline examples/repository_with_combine.rb:65]: Not the prettiest, but I got lost in trying to have a default view... there are some things related to `preload_assoc`, etc that I need to know about in order to create a default association

File: lib/rom/http/schema.rb [hunk 5]
```diff
@@ -15,6 +15,26 @@ def to_output_hash
           .schema(map { |attr| [attr.key, attr.to_read_type] }.to_h)
           .with_key_transform(&:to_sym)
       end
+
+      # To maintain compatibility with other adapters
+      #
+      # @api private
+      def qualified
```
Comment (ianks) [inline lib/rom/http/schema.rb:22]: Would y'all be OK with this?

**Comment by ianks** [comment]
Also, feel free to push up commits to this branch! Overall, would love to see this feature even if I can't get it done myself!

File: examples/repository_with_combine.rb (hunk 4 above)
Comment (solnic) [reply examples/repository_with_combine.rb:65]: @ianks how about starting with explicit views? I mean, I don't even think it's possible to just come up with default views, no?

File: examples/repository_with_combine.rb (hunk 2 above)
Comment (solnic) [reply examples/repository_with_combine.rb:46]: @ianks why is this needed?

File: lib/rom/http/schema.rb (hunk 5 above)
Comment (solnic) [reply lib/rom/http/schema.rb:22]: @ianks yes, this is fine

**Comment by ianks** [comment]
Ok @solnic I think this is ready to go, many months later lol. Would love to get this in :heart:

**Pull request merged**